### PR TITLE
Updated basic APG algorithm

### PR DESCRIPTION
**Pull request by Andrew-Luo1**
The goal of this proposed update is to provide a simple APG algorithm that can solve non-trivial tasks, as a first step for researchers and practitioners to explore Brax's differentiable simulation. It has been tested on MJX. Notes:
- A demonstration of this algorithm is shown [here](https://github.com/Andrew-Luo1/mujoco_apg_tutorial/blob/main/mjx/training_apg.ipynb). 
- I have not benchmarked this algorithm against Brax's RL algorithms such as PPO and SAC, since a) the environments would need differentiable rewards, and b) this simple implementation aims to be a basic reference to extend upon.

**1: Algorithm Update**

This fork contains an APG implementation that is about as simple as the current one, but reflects a common thread between several recent results that have used differentiable simulators to achieve locomotion: [1](https://arxiv.org/abs/2204.07137), [2](https://adaptive-horizon-actor-critic.github.io/), [3](https://arxiv.org/abs/2403.14864). 

Brax's current APG algorithm is roughly equivalent to the following pseudocode:
```
for i in range(n_epochs):
    reset state
    policy_grads = []
    for j in range(episode_length // short_horizon))
        state, policy_grad = jax.grad(unroll(state, policy, short_horizon))
        policy_grads.append(policy_grad)
    optimizer.step(mean(policy_grads))
```

In contrast, the cited results update the policy gradient much more frequently, using the observation that policy gradients that differentiate through the simulator have low variance. Hence, unrolling for an entire episode before updating has limited use. That additional samples past a certain point do not help is seen in that convergence does not increase with with massive parallelization [2]. The proposed APG algorithm essentially performs live stochastic gradient descent on the policy, unrolling it for a short window, doing a gradient update, then continuing where it left off:

```
reset state
for i in range(n_epochs):
    state, policy_grad = jax.grad(unroll(state, policy, short_horizon))
    optimizer.step(policy_grad)
```

Note that n_epochs can be much larger in this case. This modification allows the algorithm to relatively quickly learn quadruped locomotion, albeit with a particular training pipeline and reward design. ([Notebook](https://github.com/Andrew-Luo1/mujoco_apg_tutorial/blob/main/mjx/training_apg.ipynb))

Additional notes:
- This fork uses a learning schedule to improve training stability.
- The particular choices of Adam optimizer parameters come from [2] and significantly improve training outcomes.

**2: Supporting Updates**

*Configurable initial policy variance*: When hotstarting a policy, it benefits to explore around its induced state-space vicinity. This can be done by initializing the policy network weights small. Currently, the softplus disables this possibility, so this fork adds a scaling parameter.

*Layer norm*: I have found that using [layer normalization](https://arxiv.org/abs/1607.06450) in the policy neural network has greatly improved the training stability of APG methods and is seen in [other implementations](https://github.com/NVlabs/DiffRL/blob/a4c0dd1696d3c3b885ce85a3cb64370b580cb913/models/actor.py#L32).

### CODE REVIEW

File: brax/training/distribution.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ def forward_log_det_jacobian(self, x):
 class NormalTanhDistribution(ParametricDistribution):
   """Normal distribution followed by tanh."""
 
-  def __init__(self, event_size, min_std=0.001):
+  def __init__(self, event_size, min_std=0.001, var_scale=1):
```
Comment (erikfrey) [inline brax/training/distribution.py:134]: nit: add var_scale to Args section of docstring

**Review by erikfrey — CHANGES_REQUESTED** [review]
Looks great, just one nit.

**Comment by erikfrey** [comment]
Oh - it looks like you need to update [brax/training/agents/apg/train_test.py](https://github.com/google/brax/blob/main/brax/training/agents/apg/train_test.py) - changes should hopefully be minimal - please review the [failing test](https://github.com/google/brax/actions/runs/8695501482/job/23891040303?pr=476).

Also, please do sign the CLA.  Thank you!

**Comment by Andrew-Luo1** [comment]
Hi @erikfrey, I have updated the tests and they pass on my local setup. I've also fixed the nit. I've signed the CLA, and the Checks tab is saying that my signing went through. Please let me know if there's anything missing.

File: brax/training/agents/apg/train_test.py [hunk 2]
```diff
@@ -22,6 +22,9 @@
 from brax.training.agents.apg import networks as apg_networks
 from brax.training.agents.apg import train as apg
 import jax
+from jax import config
+config.update("jax_enable_x64", True)
```
Comment (erikfrey) [inline brax/training/agents/apg/train_test.py:26]: these config changes are process-wide, so it's breaking other tests that expect default float width and precision.  we run all the tests in a single process.

the envs in this test are simple enough that hopefully they don't need these config changes.  can you try removing these config changes or otherwise tweak the tests so that `jax_enable_64` and the precision change are not needed?

**Review by erikfrey — CHANGES_REQUESTED** [review]
Almost there!

**Comment by Andrew-Luo1** [comment]
I removed the double precision toggle and the tests still run fine on my local setup. Let's see if this works :)

**Comment by erikfrey** [comment]
Amazing, thank you!
